src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -1086,10 +1088,12 @@ def _get_authorization_code_worker(authority_url, resource, results):
         logger.warning("Error: can't reserve a port for authentication reply url")
         return
 
+    request_state = ''.join(random.SystemRandom().choice(string.ascii_lowercase + string.digits) for _ in range(20))
```

Comment:
Suggest capture `NotImplementedError` and fall back to the old `code`. I believe major OSs all offer the random source, but the [API](https://docs.python.org/3/library/random.html#random.SystemRandom) spec does call out not all systems support it.

Reply:
Fixed in a8bb385